internal/pkg/workspace/workspace.go

```diff
@@ -109,79 +113,120 @@ func (ws *Workspace) Summary() (*archer.WorkspaceSummary, error) {
 		if err != nil {
 			return nil, err
 		}
-		wsSummary := archer.WorkspaceSummary{}
+		wsSummary := Summary{}
 		return &wsSummary, yaml.Unmarshal(value, &wsSummary)
 	}
 	return nil, &ErrNoProjectAssociated{}
 }
 
-func (ws *Workspace) writeSummary(projectName string) error {
-	summaryPath, err := ws.summaryPath()
+// AppNames returns the application names in the workspace.
+func (ws *Workspace) AppNames() ([]string, error) {
+	projectPath, err := ws.projectDirPath()
 	if err != nil {
-		return err
+		return nil, err
 	}
+	files, err := ws.fsUtils.ReadDir(projectPath)
+	if err != nil {
+		return nil, fmt.Errorf("read directory %s: %w", projectPath, err)
+	}
+	var names []string
+	for _, f := range files {
+		if !f.IsDir() {
+			continue
+		}
+		names = append(names, filepath.Base(f.Name()))
+	}
+	return names, nil
+}
 
-	workspaceSummary := archer.WorkspaceSummary{
-		ProjectName: projectName,
+// Read returns the contents of the file under the project directory joined by path elements.
+func (ws *Workspace) Read(elem ...string) ([]byte, error) {
+	projectPath, err := ws.projectDirPath()
+	if err != nil {
+		return nil, err
 	}
+	pathElems := append([]string{projectPath}, elem...)
+	return ws.fsUtils.ReadFile(filepath.Join(pathElems...))
+}
 
-	serializedWorkspaceSummary, err := yaml.Marshal(workspaceSummary)
+// Write writes the data to a file under the project directory joined by path elements.
+// If successful returns the full path of the file, otherwise returns an empty string and the error.
+func (ws *Workspace) Write(data []byte, elem ...string) (string, error) {
```

Comment:
This looks super nice. I think, though, that I'd still like a higher level func for `WriteManifest(manifestBytes[], appName)` and eventually `WriteAppAddons(...)` just so no one outside of the workspace struct needs to know the layout. What do you think?

Reply:
jes, that makes sense!
Added the following:
* `ReadAppManifest`
* `ReadPipelineManifest`
* `WriteAppManifest`
* `WritePipelineBuildspec`
* `WritePipelineManifest`

Reply:
oh also part of this change, I removed the `--pipeline-file` flag from `pipeline_update.go` if you'd like to take a look at that.